src/middleware/issueDetails.middleware.js

```diff
@@ -146,129 +68,68 @@ const getIssueField = (text, html, classes) => {
-  if (issuesByEntryNumber[entryNumber]) {
-    issueIndex = issuesByEntryNumber[entryNumber].findIndex(
-      (issue) => issue.field === row.field
-    )
-    hasError = issueIndex >= 0
-  }
+export const setPagePaginationOptions = (req, res, next) => {
+  const { issueEntitiesCount } = req
+  const { lpa, dataset: datasetId, issue_type: issueType, issue_field: issueField } = req.params
 
-  let valueHtml = ''
-  let classes = ''
-  if (hasError) {
-    const message =
-        issuesByEntryNumber[entryNumber][issueIndex].message || issueType
-    valueHtml += issueErrorMessageHtml(message, null)
-    classes += 'dl-summary-card-list__row--error'
-  }
-  valueHtml += row.value
+  req.resultsCount = issueEntitiesCount
+  req.urlSubPath = `/organisations/${lpa}/${datasetId}/${issueType}/${issueField}/entry/`
+  req.paginationPageLength = 1
 
-  return getIssueField(row.field, valueHtml, classes)
+  next()
 }
 
-/***
- * Middleware. Updates req with `templateParams`
+/**
+ * Middleware. Prepares template parameters for the issue details page.
+ *
+ * @param {object} req - The request object
+ * @param {object} res - The response object (not used)
+ * @param {function} next - The next middleware function
+ *
+ * @summary Extracts relevant data from the request and organizes it into a template parameters object.
+ * @description This middleware function prepares the template parameters for the issue details page.
+ * It extracts the entry data, issue entities count, issues by entry number, error summary, and other relevant data
+ * from the request, and organizes it into a template parameters object that can be used to render the page.
  */
 export function prepareIssueDetailsTemplateParams (req, res, next) {
-  const { entryData, pageNumber, issueEntitiesCount, issuesByEntryNumber, entryNumber, entityCount: entityCountRow } = req
-  const { lpa, dataset: datasetId, issue_type: issueType, issue_field: issueField } = req.params
-  const { entity_count: entityCount } = entityCountRow ?? { entity_count: 0 }
-
-  let errorHeading
-  let issueItems
-
-  const BaseSubpath = `/organisations/${lpa}/${datasetId}/${issueType}/${issueField}/`
-
-  if (Object.keys(issuesByEntryNumber).length < entityCount) {
-    errorHeading = performanceDbApi.getTaskMessage({ issue_type: issueType, num_issues: issueEntitiesCount, entityCount, field: issueField }, true)
-    issueItems = Object.entries(issuesByEntryNumber).map(([entryNumber, issues], i) => {
-      const pageNum = i + 1
-      return {
-        html: performanceDbApi.getTaskMessage({ issue_type: issueType, num_issues: 1, field: issueField }) + ` in record ${entryNumber}`,
-        href: `${BaseSubpath}${pageNum}`
+  const { entities, issueEntitiesCount, errorSummary, specification, pagination } = req
+  const { issue_type: issueType, issue_field: issueField, pageNumber: pageNumberString } = req.params
+  const pageNumber = parseInt(pageNumberString)
+
+  const entity = entities[pageNumber - 1]
```

Comment:
If schema is stricter (see previous comment), validation of `pageNumber` is taken care of.  But we still need to check if `entities[pageNumber -1] is within bounds. I think it would be fine to short circuit with 400 error if not.